File: renderer/fov.py
from math import cos, sin, radians
from dda import cast_ray_dda

def cast_fov(grid, cols, rows, tile_size, ox, oy, cam_angle, fov_deg, num_rays):
    fov   = radians(fov_deg)
    start = cam_angle - fov / 2
    step  = fov / (num_rays - 1)

    hits_of_each_ray = []

    for i in range(num_rays):
        ang = start + i * step
        dx  = cos(ang)
        dy  = sin(ang)
        hits_of_each_ray.append(cast_ray_dda(grid, cols, rows, tile_size, ox, oy, dx, dy))

    return hits_of_each_ray

File: game/fov.go
package game

import "math"

// castFOV fires NumRays across FOVDegrees centred on camAngle and returns
// each ray's hit list, left to right.
func castFOV(m *Map, ox, oy, camAngle float64) [][]rayHit {
	fov := float64(FOVDegrees) * math.Pi / 180
	start := camAngle - fov/2
	step := fov / float64(NumRays-1)

	hits := make([][]rayHit, NumRays)
	for i := range hits {
		ang := start + float64(i)*step
		hits[i] = castRayDDA(m, ox, oy, math.Cos(ang), math.Sin(ang))
	}
	return hits
}
